server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -480,7 +480,18 @@ public <T> T getTransient(String key) {
      * @param value       the header value
      */
     public void addResponseHeader(final String key, final String value) {
-        addResponseHeader(key, value, v -> v);
+        addResponseHeader(key, value, v -> v, false);
+    }
+
+    /**
+     * Add the {@code value} for the specified {@code key}
+     *
+     * @param key         the header name
+     * @param value       the header value
+     * @param replaceExistingKey whether to replace the existing header if it already exists
+     */
+    public void addResponseHeader(final String key, final String value, final boolean replaceExistingKey) {
+        addResponseHeader(key, value, v -> v, replaceExistingKey);
```

Comment:
I'm wondering if it's better to have a separate `update/replaceResponseHeader` method instead as this is a little semantically confusing to me

Reply:
Make sense. let me refactor this part